### PR TITLE
Fix/icl 175/upgrade dependencies

### PR DESCRIPTION
**Pull request by jrhender**
Object of this PR is to try to reduce the vulnerabilities that show up when scanning the `iam-contracts` npm package. See https://energyweb.atlassian.net/browse/IS-2

A few techniques are used:
- `--no-optional` when installing packages
- moved several packages to `devDependencies`

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -15,7 +15,8 @@ jobs:
           registry-url: https://registry.npmjs.org/
 
       - name: Install dependencies
-        run: npm install
+        # Using --no-optional to reduce vulnerabilities
+        run: npm install --no-optional
```
Comment (Harasz) [inline .github/workflows/pull-request.yml:19]: Is there a particular reason why we do not use `npm ci` here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -28,7 +28,8 @@ jobs:
           registry-url: https://registry.npmjs.org/
 
       - name: Install dependencies
-        run: npm install
+        # Using --no-optional to reduce vulnerabilities
+        run: npm install --no-optional
```
Comment (Harasz) [inline .github/workflows/release.yml:32]: Also here

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (jrhender) [reply .github/workflows/pull-request.yml:19]: Good point, I will change to be `npm ci --no-optional`

**Review by Harasz — APPROVED** [review]

**Comment by nichonien** [comment]
@jrhender The changes looks good, though the story linked should be [ICS-144](https://energyweb.atlassian.net/browse/ICS-144?atlOrigin=eyJpIjoiMjhhMmZlN2VkNTZiNDc3MWI5NTMzMmU1MzYxZDU2YzQiLCJwIjoiaiJ9) . Also, this PR still doesn't fix the `vulnerabilities`. This could be a separate story maybe, What are your thoughts?

**Comment by jrhender** [comment]
@nichonien when I run `npm audit --production` locally, I get 0 vulnerabilities. So I think that should be sufficient for AEMO's security scanning, though I could be wrong. In my opinion, we wouldn't need another story to further resolve the vulnerabilities. What do you think?

**Pull request merged**